src/main/java/com/avispl/dal/communicator/polycom/videoos/PolycomVideoOS.java

```diff
@@ -0,0 +1,1046 @@
+    @Override
+    public String dial(DialDevice dialDevice) throws Exception {
+        if (logger.isDebugEnabled()) {
+            logger.debug("Dialing using dial string: " + dialDevice.getDialString());
+        }
+        checkAndInitialize();
+        ObjectNode request = JsonNodeFactory.instance.objectNode();
+        Integer callSpeed = dialDevice.getCallSpeed();
+
+        request.put("address", dialDevice.getDialString());
+        request.put("rate", (callSpeed != null && callSpeed > 0) ? callSpeed : DEFAULT_CALL_RATE);
+
+        Protocol protocol = dialDevice.getProtocol();
+        if (protocol != null) {
+            request.put("dialType", protocol.name());
+        }
+        ArrayNode response = doPost(CONFERENCE, request, ArrayNode.class);
+        if (response == null) {
+            // Valid response was not received, cannot go further to checking the conference id retrieval
+            throw new IllegalStateException(String.format("Unable to receive response from %s", CONFERENCE));
+        }
+
+        String meetingInfoUrl = getJsonProperty(response.get(0), "href", String.class);
+        for (int i = 0; i < MAX_STATUS_POLL_ATTEMPT; i++) {
+            JsonNode meetingInfo = doGet(meetingInfoUrl, JsonNode.class);
+            if (meetingInfo != null) {
+                String conferenceId = getJsonProperty(meetingInfo, "parentConfId", String.class);
+                if (null != conferenceId) {
+                    String remoteAddress = getJsonProperty(meetingInfo, "address", String.class);
+                    if (!StringUtils.isNullOrEmpty(remoteAddress) && remoteAddress.trim().equals(dialDevice.getDialString().trim())) {
+                        return conferenceId;
+                    }
+                }
+            }
+            Thread.sleep(1000);
+        }
+        return null;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * Locking is necessary because statistics is still being collected there and if the device is in the call
+     * and collecting some information available while in the call only - it'll end up with a 404 error.
+     * If there's a conferenceId available - only one conference is removed, otherwise - the method
+     * iterates through all of the available conferences and removes them.
+     */
+    @Override
+    public void hangup(String conferenceId) throws Exception {
+        if (logger.isDebugEnabled()) {
+            logger.debug("Hangup string received: " + conferenceId);
+        }
+        checkAndInitialize();
+        controlOperationsLock.lock();
+        try {
+            if (!StringUtils.isNullOrEmpty(conferenceId)) {
+                doDelete(String.format(CONFERENCES, conferenceId));
+            } else {
+                ArrayNode conferenceCall = listConferenceCalls();
+                for (JsonNode node : conferenceCall) {
+                    Integer id = getJsonProperty(node, "id", Integer.class);
+                    doDelete(String.format(CONFERENCES, id));

```

Comment:
Similar to comment above - it is not obvious why it is deleing conferences instead of just hanging up from the call as asked, and it is not documented.

Reply:
Adding VideoOS REST API doc link + javadocs explaining this.